public/src/components/campaign/joined_campaigns.jsx

```diff
@@ -9,15 +9,27 @@ export default class JoinedCampaigns extends Component {
   componentDidMount() {
     const { id } = this.props.auth;
     if (id) {
+      this.props.clearCampaigns();
       this.props.fetchCampaignsByUser(id);
     }
   }
 
   render() {
-    const { history, joined_campaigns, current_campaign, setCurrentCampaign } = this.props;
+    const { history, joined_campaigns, setCurrentCampaign } = this.props;
     const { first_name, last_name } = this.props.account_info;
     const page = 'Campaign';
     const is_admin = false;
+
+    let { current_campaign } = this.props;
+    if (
+      joined_campaigns &&
+      current_campaign &&
+      Object.getOwnPropertyNames(joined_campaigns).length > 0 &&
```

Comment:
I think .getOwnPropertyNames is more reliable for the purposes of checking to see if an object is empty or not, because .keys will only return enumerable properties.